teams_list.go

```diff
@@ -106,10 +116,24 @@ func (api *API) TeamsList(ctx context.Context, accountID, listID string) (TeamsL
 // TeamsListItems returns all list items for a list.
 //
 // API reference: https://api.cloudflare.com/#teams-lists-teams-list-items
-func (api *API) TeamsListItems(ctx context.Context, accountID, listID string) ([]TeamsListItem, ResultInfo, error) {
-	uri := fmt.Sprintf("/%s/%s/gateway/lists/%s/items", AccountRouteRoot, accountID, listID)
+func (api *API) TeamsListItems(ctx context.Context, params TeamsListItemsParams) ([]TeamsListItem, ResultInfo, error) {
+	if params.AccountID == "" {
+		return []TeamsListItem{}, ResultInfo{}, ErrMissingAccountID
+	}
 
-	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if params.ListID == "" {
+		return []TeamsListItem{}, ResultInfo{}, ErrMissingListID
+	}
+
+	v, _ := query.Values(params)
```

Comment:
shouldn't this only be referencing `params.PaginationOptions`?  otherwise, won't it include the `AccountId` and `ListId` fields too and produce incorrect query parameters?

Reply:
no, internally, we use https://github.com/google/go-querystring for query parameter marshalling based on struct field tags. using `url:"-"` allows defining the field but ignoring it for the purposes of building the URL. see https://github.com/google/go-querystring/blob/v1.1.0/query/encode.go#L48-L52 for the full details.

the second parameter to `TeamsListItems` is `TeamsListItemsParams` as per the [experimental client documentation](https://github.com/cloudflare/cloudflare-go/blob/master/docs/experimental.md) where we are making the move to for all methods, eventually.